src/blockchain/ain/node/node-client.ts

```diff
@@ -48,6 +48,7 @@ export class NodeClient {
   }
 
   async getBlock(hash: string): Promise<Block<string>> {
+    if (!hash) return Promise.reject();
```

Comment:
Add a reason to improve logging, maybe change to throwing:
`if (!hash) throw new Error('Hash undefined');`
or similar.

Reply:
Same below